process/indexer/outportBlockCache.go

```diff
@@ -0,0 +1,68 @@
+package indexer
+
+import (
+	"encoding/hex"
+	"fmt"
+	"sync"
+
+	"github.com/multiversx/mx-chain-core-go/data/outport"
+)
+
+type outportBlockCache struct {
+	cache      map[string]*outport.OutportBlock
+	cacheMutex sync.RWMutex
+}
+
+// NewOutportBlockCache creates a new LRU cache able to store *outport.OutportBlock
+func NewOutportBlockCache() *outportBlockCache {
+	return &outportBlockCache{
+		cache:      make(map[string]*outport.OutportBlock),
+		cacheMutex: sync.RWMutex{},
+	}
+}
+
+// Add will add the block to internal cache, if not nil and if the hash doesn't already exist
+// If full, it will evict the oldest entry in cache
+func (obc *outportBlockCache) Add(outportBlock *outport.OutportBlock) error {
+	obc.cacheMutex.Lock()
+	defer obc.cacheMutex.Unlock()
+
+	if outportBlock == nil || outportBlock.BlockData == nil {
```

Comment:
This can be moved outside of the mutex-protected area

Reply:
Right, refactored here